### PR TITLE
Add exchange items to feed and feed item detail

### LINKED ISSUE #1595: Support CELO purchases/sells in the new transaction feed.
In the work added here: https://github.com/valora-inc/wallet/pull/1575
we didn't add support for Exchanges in the Transaction Feed. We need to add an item to the feed when an exchange is made.
It's not clear to me what this should look like exactly, so maybe sync with Nitya to confirm.

We also need to add a standby transaction when an exchange is made.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Added feed items and details for CELO purchases and sells.

Screen Shot 2021-12-30 at 20 44 33
Screen Shot 2021-12-30 at 20 44 51

Updated video after changes:

https://user-images.githubusercontent.com/6062888/148303075-a1fa7307-1cee-49a0-ab89-cec5a1c11590.mov

### Other changes

N/A

### Tested

Manually and with unit tests

### How others should test

With the feed multi token support flag enabled, buy/sell CELO and see that the item shows correctly in the feed. 
Then click it and see that the transaction detail looks correct as well.

### Related issues

- Fixes #1595

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.test.tsx [hunk 1]
```diff
@@ -144,4 +183,42 @@ describe('TransactionDetailsScreen', () => {
     const totalComponent = getByTestId('TotalLineItem/Total')
     expect(getElementText(totalComponent)).toEqual('₱13.30')
   })
+
+  it('renders correctly for CELO purchases', async () => {
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/TransactionDetailsScreen.test.tsx:187]: Should we add the `sold CELO` case?

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.test.tsx [hunk 2]
```diff
@@ -144,4 +183,42 @@ describe('TransactionDetailsScreen', () => {
...
+
+    const celoAmount = getByTestId('CeloAmount')
+    expect(getElementText(celoAmount)).toEqual('3.00')
+
+    const fiatAmount = getByTestId('FiatAmount')
+    expect(getElementText(fiatAmount)).toEqual('₱13.30')
+
+    // Includes the fee
+    const total = getByTestId('TotalLineItem/Total')
+    expect(getElementText(total)).toEqual('₱13.43')
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/TransactionDetailsScreen.test.tsx:219]: Can we check the fee drawer amount too?

**Review by jeanregisser — COMMENTED** [review]
Great! 🚀 
I'm just confused by the screenshots:
- one feed item shows "Purchased CELO -€0.071" for 0.01 CELO
- but the corresponding details shows it as €0.042 for 0.01 CELO and the total doesn't add up with the fees.

**Comment by gnardini** [comment]
Yup, so there are different errors here.

- This is an exchange, so the CELO and cUSD amounts should be equal, but they are not.
- We were not using the fiat "currency info" coming from the blockchain-api in one place.

The fee thing I'm pretty sure it's a rounding issue. If we want to avoid it we need to show more decimals :/

I agree this is confusing though, I'll take a closer look at it to see how I can make it work more as expected

**Comment by gnardini** [comment]
I confirmed that the issue is that the CELO price is wrong. I switched the UI to use the stable token value everywhere for consistency except in the exchange rate, but this way at least the feed item matches with the detail screen. 

Regarding the fee, in this case it was `0.0000626925 CELO` which the `FeeDrawer` rounds up to `0.001`. The `FeeDrawer` component is still the old one, hasn't been touched in a while, so it's not using the rounding rules from `TokenDisplay`, maybe something we can work on in the future? If the value was higher it should have been added correctly to the total.

**Comment by gnardini** [comment]
I updated the description with a video. In the video we can see that for the Sold CELO case because of rounding the fee adds a cent to the total, but for purchasing CELO it doesn't. Not sure how to fix that in this issue and I think that's a bigger issue we have around rounding throughout the app

**Comment by jeanregisser** [comment]
Sorry I don't see the video in the PR description.

**Review by jeanregisser — APPROVED** [review]

**Comment by gnardini** [comment]
We also deployed blockchain-api to mainnet and the exchange rate and values match there and show up properly. It's probably an issue caused by the crazy arbitrage opportunities in Alfajores, exchange rates are not consistent there :/

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified this task on latest Android Internal Build V: 1.24.1 and Latest iOS Test Flight Build V: 1.24.1 and updated the status in Gsheet 

**Sheet Link:** https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=767055179

**Observation :** "0" amount is seen for fees section on CELO page whereas "0.001" amount is seen for fees section for same purchased / sold CELO activity and we have raised [query](https://github.com/valora-inc/wallet/issues/1802) for it 

**Devices :** Google Pixel 2XL(11.0) , iPhone 12 (14.7.1)

**CC:** @gnardini

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified this task on latest Android Internal Build V: 1.24.1 and Latest iOS Test Flight Build V: 1.24.1 and updated the status in Gsheet

**Sheet Link**: https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=767055179

**Observation** : "0" amount is seen for fees section on CELO page whereas "0.001" amount is seen for fees section for same purchased / sold CELO activity and we have raised [query](https://github.com/valora-inc/wallet/issues/1802) for it

**Devices** : Samsung Galaxy M40 (10.0)

CC: @gnardini 

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this task on latest Android Internal Release Build V:1.24.1 and updated the status in Gsheet.

**Sheet Link**: https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=767055179

**Observation** : "0" amount is seen for fees section on CELO page whereas "0.001" amount is seen for fees section for same purchased / sold CELO activity and we have raised [query](https://github.com/valora-inc/wallet/issues/1802) for it

**Bug:** 
[Link1](https://github.com/valora-inc/wallet/issues/1746)
[Link2](https://github.com/valora-inc/wallet/issues/1773)

**Devices:** Redmi Note 5 Pro (9.0)

CC: @gnardini 
Thanks..!!

**Pull request merged**